presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/ThriftExceptions.java

```diff
@@ -37,7 +37,7 @@ public static PrestoException toPrestoException(Exception e)
             throw new PrestoException(THRIFT_SERVICE_NO_AVAILABLE_HOSTS, e);
         }
         if ((e instanceof TApplicationException) || (e instanceof PrestoThriftServiceException)) {
-            return new PrestoException(THRIFT_SERVICE_GENERIC_REMOTE_ERROR, "Exception raised by remote Thrift server", e);
+            return new PrestoException(THRIFT_SERVICE_GENERIC_REMOTE_ERROR, "Exception raised by remote Thrift server. Original error message: " + e.getMessage(), e);
```

Comment:
"Exception raised by remote Thrift server: " + e.getMessage()
would be more concise?